docs/advanced/thunks.md

```diff
@@ -0,0 +1,37 @@
+The `thunkmiddleware` packaged with Rodux will intercept any `action` dispatched to your `store` that is a Lua *function* and simply execute that function instead of forwarding your `action` to the `reducer`. These functions (also called thunks) have access to the `store` and are allowed to dispatch `action` objects as necessary.
+
+!!! info
+	the following examples assume that you've successfully [installed Rodux](installation.md) into `ReplicatedStorage`!
+
+```lua
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+
+local Rodux = require(ReplicatedStorage.Rodux)
+
+local reducer = function(state, action)
+	-- reducer that handles actions of type "MadeANewFriend"
+end
+
+local initialState = {}
+
+local store = Rodux.Store.new(reducer, initialState, {
+	Rodux.thunkMiddleware,
+})
+
+--[[
+	Our thunkMiddleware will process this action as a thunk
+	since it is a Lua function
+]]
+store:dispatch(function(store)
+	getAsyncFriendsForUser("Sarah", function(result)
+		for _, friend in ipairs(result)
+			store:dispatch({
+				type = "MadeANewFriend",
+				friend = friend,
+			})
```

Comment:
I had the same thought, but was trying to go for simplicity rather than a fully robust solution. Agreed though, it's misleading, I'll rewrite it so it's done in a more production ready way.